CurrencyConverterControl/CurrencyConverterControl.cs

```diff
@@ -0,0 +1,251 @@
+            if (comboBox != null && comboBox.Items.Count > 0)
+            {
+                comboBox.SelectedIndex = 0; 
+            }
+        }
+
+        // Perform the convertion 
+        private void CmbSource_SelectionChanged(object sender, SelectionChangedEventArgs args)
+        {
+            try
+            {
+                //check if all have values before going to API calculate
+                if (SourceCurrency is null|| String.IsNullOrEmpty(SourceCurrency.CurrencyCode) 
+                   || DestinationCurrency is null || String.IsNullOrEmpty(DestinationCurrency.CurrencyCode))
+                {
+                    return;
+                }
+                OutputValue = CurrencyDataProvider.ConvertAsync(SourceCurrency.CurrencyCode, DestinationCurrency.CurrencyCode, InputValue).Result;
+                _currencyDestination.SelectedValue = DestinationCurrency;
+            }
+            catch (Exception ex) { Console.WriteLine($"Error: {ex.Message}"); }
+        }
+        private void CmbDestination_SelectionChanged(object sender, RoutedEventArgs args)
+        {
+            try
+            {
+                //check if all have values before going to API calculate
+                if (SourceCurrency is null || String.IsNullOrEmpty(SourceCurrency.CurrencyCode)
+                   || DestinationCurrency is null || String.IsNullOrEmpty(DestinationCurrency.CurrencyCode))
+                {
+                    MessageBox.Show("Please fill in all fields!", "Warning", MessageBoxButton.OK, MessageBoxImage.Warning);
+                    return;
+                }
+                OutputValue = CurrencyDataProvider.ConvertAsync(SourceCurrency.CurrencyCode, DestinationCurrency.CurrencyCode, InputValue).Result;
+                _currencySource.SelectedValue = SourceCurrency;
+                
+            }
+            catch (Exception ex) { Console.WriteLine($"Error: {ex.Message}"); }
+        }
+
+
+        #endregion
+
+        #region Notify Property Change 
+        public event PropertyChangedEventHandler PropertyChanged;
+        protected virtual void RaisePropertychanged([CallerMemberName] string propertyName = null)
+        {
+            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
+        }
+
+        public virtual Task LoadAsync() 
+        {
+            // Call the protected LoadCurrencies method
+            LoadCurrencies();
+            return Task.CompletedTask;
+        }
+        #endregion
+        #region public functions
+        /// <summary>
+        /// This function is used to execute the convertion
+        /// of an amount from USD to the chosen currency 
+        /// </summary>
+        /// <param name="currencyFrom"></param>
+        /// <param name="currencyTo"></param>
+        /// <param name="amount"></param>
+        /// <returns>Returns the convertion value</returns>
+        public double ConvertValues(string currencyFrom, string currencyTo, double amount)
```

Comment:
I totally agree. 
My point is why other public API uses Currency? They also can use string.
The user at the control level should not care about the flag of the currency, it can be handled at the provider level (but, in my opinion, this should be internal and completely under the currency provider responsibility).